src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -411,6 +405,19 @@ public void deleteDataset(Integer datasetId, Integer userId) throws Exception {
         ).collect(Collectors.toList());
     }
 
+    private boolean filterDatasetOnProperties(DatasetDTO dataset, String term) {
+        //datasets need to have consentId, null check to prevent NPE
+        String consentId = dataset.getConsentId();
+        Boolean consentIdMatch = Objects.nonNull(consentId) && consentId.toLowerCase().contains(term);
+        return dataset.getProperties()
+            .stream()
+            .filter(p -> Objects.nonNull(p.getPropertyValue()))
+            .anyMatch(p -> {
+                return consentIdMatch
+                || p.getPropertyValue().toLowerCase().contains(term);
+            });
```

Comment:
Minor optimization can be made by pulling `consentIdMatch ||` out into the front of the overall return statement:
```suggestion
        return consentIdMatch || dataset.getProperties()
            .stream()
            .filter(p -> Objects.nonNull(p.getPropertyValue()))
            .anyMatch(p -> {
                return p.getPropertyValue().toLowerCase().contains(term);
            });
```